tools/cnvkit-batch.cwl

```diff
@@ -0,0 +1,173 @@
+                reference will be built.
+  inputBinding:
+    prefix: --normal 
+
+- id: fasta
+  type: ["null", string]
+  description: Reference genome, FASTA format (e.g. UCSC hg19.fa)
+  inputBinding:
+    prefix: --fasta 
+
+- id: targets
+  type: ["null", string]
+  description: Target intervals (.bed or .list)
+  inputBinding:
+    prefix: --targets 
+
+- id: antitargets
+  type: ["null", string]
+  description: Antitarget intervals (.bed or .list)
+  inputBinding:
+    prefix: --antitargets 
+
+- id: annotate
+  type: ["null", string]
+  description: UCSC refFlat.txt or ensFlat.txt file for the reference genome.
+                Pull gene names from this file and assign them to the target
+                regions.
+  inputBinding:
+    prefix: --annotate 
+
+- id: short_names
+  type: ["null", boolean]
+  default: null
+  description: Reduce multi-accession bait labels to be short and consistent.
+  inputBinding:
+    prefix: --short-names 
+
+- id: split
+  type: ["null", boolean]
+  default: null
+  description: Split large tiled intervals into smaller, consecutive targets.
+  inputBinding:
+    prefix: --split 
+
+- id: target_avg_size
+  type: ["null", int]
+  description: Average size of split target bins (results are approximate).
+  inputBinding:
+    prefix: --target-avg-size 
+
+- id: access
+  type: ["null", string]
+  description: Regions of accessible sequence on chromosomes (.bed), as
+                output by the 'access' command.
+  inputBinding:
+    prefix: --access 
+
+- id: antitarget_avg_size
+  type: ["null", int]
+  description: Average size of antitarget bins (results are approximate).
+  inputBinding:
+    prefix: --antitarget-avg-size 
+
+- id: antitarget_min_size
+  type: ["null", int]
+  description: Minimum size of antitarget bins (smaller regions are dropped).
+  inputBinding:
+    prefix: --antitarget-min-size 
+
+- id: output_reference
+  type: ["null", string]
+  description: Output filename/path for the new reference file being created.
+                (If given, ignores the -o/--output-dir option and will write the
+                file to the given path. Otherwise, "reference.cnn" will be
+                created in the current directory or specified output directory.)
+                
+  inputBinding:
+    prefix: --output-reference 
+
+- id: reference
+  type: ["null", string]
+  description: Copy number reference file (.cnn).
+  inputBinding:
+    prefix: --reference
+
+- id: output_dir
+  type: ["null", string]
+  default: .
+  description: Output directory.
+  inputBinding:
+    prefix: --output-dir 
+
+- id: scatter
+  type: ["null", boolean]
+  default: null
+  description: Create a whole-genome copy ratio profile as a PDF scatter plot.
+  inputBinding:
+    prefix: --scatter
+
+- id: diagram
+  type: ["null", boolean]
+  default: null
+  description: Create a diagram of copy ratios on chromosomes as a PDF.
+  inputBinding:
+    prefix: --diagram 
+
+outputs:
+    []
```

Comment:
There are several outputs from this command and they vary based on the input BAM filenames and the options given.

- For each tumor/test-sample BAM named e.g. Sample.bam, the outputs are: "Sample.targetcoverage.cnn", "Sample.antitargetcoverage.cnn", "Sample.cnr", "Sample.cns"
- If the `--scatter` option is given, then for each tumor/test sample, "Sample-scatter.pdf" is created
- Similarly, the `--diagram` option creates "Sample-diagram.pdf"
- For all of the above, if `-d/--output-dir` is specified, the created file names are relative to (i.e. in) that specified directory
- If the `-r/--reference` option is *not* given, then a `.cnn` file is created either with the filename given by `--output-reference` (regardless of the `-d/--output-dir` path) or by default "cnv_reference.cnn"